decidim/decidim
# Remove to_s from Arel.sql statements

#### :tophat: What? Why?

As far as I know, the `Arel.sql` method is use to inject safe SQL inside queries. by using `to_s` after it, it becomes unsafe SQL again. Until now it didn't seem to be a problem but we found that is affecting the Proposal's scope `sort_by_valuation_assignments_count_asc/desc`.

This PR, removes all `to_s` statements (note that the other might be irrelevant as Rails might not find it dangerous), but still I think we should not use it.

Looks like this is from Decidim 0.28 onwards.

#### :pushpin: Related Issues
*Link your PR to an issue*
- Related to #?
- Fixes #?

#### Testing

1. Enter the console of any Decidim (`bin/rails c`)
2. Run the following:
  `Decidim::Proposals::Proposal.sort_by_valuation_assignments_count_asc`
3. See this error:
```
lib/active_record/sanitization.rb:145:in `disallow_raw_sql!': Query method called with non-attribute argument(s): "( SELECT COUNT(decidim_proposals_valuation_assignments.id) FROM decidim_proposals_valuation_assignments WHERE decidim_proposals_valuation_assignments.decidim_proposal_id = decidim_proposals_proposals.id GROUP BY decidim_proposals_valuation_assignments.decidim_proposal_id ) ASC NULLS FIRST" (ActiveRecord::UnknownAttributeReference)
```
4. Apply this fix
5. See it working normally

### :camera: Screenshots
*Please add screenshots of the changes you are proposing*
![Description](URL)

:hearts: Thank you!
